include/boost/multiprecision/detail/common_factor.hpp

```diff
@@ -0,0 +1,65 @@
+///////////////////////////////////////////////////////////////
+//  Copyright 2021 Matt Borland.
+//  Distributed under the Boost Software License, Version 1.0.
+//  (See accompanying file LICENSE_1_0.txt or copy at
+//   https://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_MP_COMMON_FACTOR_HPP
+#define BOOST_MP_COMMON_FACTOR_HPP
+
+#include <boost/multiprecision/detail/standalone_config.hpp>
+
+#ifndef BOOST_MP_STANDALONE
+
+#include <boost/integer/common_factor_rt.hpp>
+
+namespace boost { namespace multiprecision { namespace detail { 
+    
+template <typename T>
+inline BOOST_CXX14_CONSTEXPR T constexpr_gcd(T a, T b) noexcept
+{
+    return boost::integer::gcd(a, b);
+}
+
+template <typename T>
+inline BOOST_CXX14_CONSTEXPR T constexpr_lcm(T a, T b) noexcept
+{
+    return boost::integer::lcm(a, b);
+}
+}}} // Namespaces
+
+#else 
+
+namespace boost { namespace multiprecision { namespace detail { 
+
+template <typename T>
+inline BOOST_CXX14_CONSTEXPR void constexpr_swap(T& a, T& b) noexcept
```

Comment:
I wonder/worry a bit about this - gcd is typically time-critical and binary-gcd is usually faster than Euclid.  We also already have: https://github.com/boostorg/multiprecision/blob/d6b0f2ecd50992e868652806e437991c771dc6b2/include/boost/multiprecision/cpp_int/misc.hpp#L397-L437 and perhaps we should do something similar here and/or share code?